File: node/transcribe_file_short.js
const { SpeechClient } = require("@soniox/soniox-node");

/**
 * Set your Soniox Cloud API key:
 * from command line: export SONIOX_API_KEY=<YOUR-API-KEY>
 * or
 * pass config object: new SpeechClient({ api_key: "<YOUR-API-KEY>" })
 */
const speechClient = new SpeechClient();

(async function () {
  const result = await speechClient.transcribeFileShort(
    "../test_data/test_audio.flac"
  );

  for (const word of result.words) {
    console.log(`${word.text} ${word.start_ms} ${word.duration_ms}`);
  }
})();

File: python/transcribe_file_short.py
from soniox.transcribe_file import transcribe_file_short
from soniox.speech_service import SpeechClient


def main():
    with SpeechClient() as client:
        result = transcribe_file_short(
            "../test_data/test_audio.flac",
            client,
            model="en_v2",
        )
        print(f"Text: " + "".join(word.text for word in result.words))
        print("Tokens:")
        for word in result.words:
            print(f"    '{word.text}' {word.start_ms} {word.duration_ms}")


if __name__ == "__main__":
    main()
